misc/toml-parser/src/main/java/io/ballerina/toml/validator/schema/Schema.java

```diff
@@ -72,16 +77,30 @@ public static Schema from(Path jsonPath) throws IOException {
     public static Schema from(String jsonContent) {
         Gson gson = new GsonBuilder()
                 .registerTypeAdapter(AbstractSchema.class, new SchemaDeserializer()).create();
-        Schema rootSchema = gson.fromJson(jsonContent, Schema.class);
-        rootSchema.setType(Type.OBJECT);
-        return rootSchema;
+        return (Schema) gson.fromJson(jsonContent, AbstractSchema.class);
```

Comment:
Seems like this casting is unnecessary

Reply:
Casting is needed as the static method should return a ObjectSchema(Schema) instead of AbstractSchema. 
Note in the old code we called `gson.fromJson(reader, Schema.class);` and now we call `gson.fromJson(reader, AbstractSchema.class);`